packages/nuxt/src/components/runtime/client-delayed-component.ts

```diff
@@ -0,0 +1,56 @@
+import { defineComponent, h, onBeforeUnmount, onMounted, ref } from 'vue'
+import type { Component, Ref } from 'vue'
+// import ClientOnly from '#app/components/client-only'
+import { useObserver } from '#app/utils'
+
+/* @__NO_SIDE_EFFECTS__ */
+export const createLazyIOClientPage = (componentLoader: Component) => {
+  return defineComponent({
+    inheritAttrs: false,
+    setup (_, { attrs }) {
+      const isIntersecting = ref(false)
+      const el: Ref<Element | null> = ref(null)
+      let unobserve: (() => void) | null = null
+      onMounted(() => {
+        const observer = useObserver()
+        unobserve = observer!.observe(el.value as Element, () => {
+          isIntersecting.value = true
+          unobserve?.()
+          unobserve = null
+        })
+      })
+      onBeforeUnmount(() => {
+        unobserve?.()
+        unobserve = null
+      })
+      return () => h('div', { ref: el }, [
+        isIntersecting.value ? h(componentLoader, attrs) : null,

```

Comment:
Do we want to lazy hydrate or to lazy load here ?

Lazy hydration would need to render the component in SSR. Keep it "static"/non-interactive in client until loaded

Reply:
Yea, we probably want all of these wrappers to lazy load as opposed to lazy hydrate. Can this be lazy loaded via 
```ts
h(await import(compName))
```?

Reply:
It's more about how `createClientIO` should work.

The issue with the current state is that it is a client-only until observed because it is never rendered in SSR.  So when missused end users can experience a reduced UX (like when using `CLientOnly`). And also having an reduced webperf score if the lazyhydrated component is within the viewport at loading.

I think what #24242 was expecting is to render the component in SSR and then hydrate when observed.

So we could probably render the component in server side. Then until observed, it would be a static vnode (you can check how to do it in nuxt-island). Once observed, you'll be able to render the component.

I think the PR probably just need some changes within `createClientIO`

Reply:
Perhaps something along the lines of:
```ts
return () => h('div', { ref: el }, [
   isIntersecting.value ? h(defineAsyncComponent(import('./Comp.vue').then(c=>c.default || c)), attrs) : null
])
```

Reply:
> The issue with the current state is that it is a client-only until observed because it is never rendered in SSR. So when missused, end users can experience a reduced UX (like when using CLientOnly). And also having an reduced webperf score if the lazyhydrated component is within the viewport at loading.
Though the question here would be, would lazy loading not inherently provide better performance as opposed to lazy hydration? We can use a static vnode of course, but wouldn't lazy loading still reduce bundle? Of course everything can be misused, it's up to the individual developers to use the tools as intended..

Reply:
It's more about the meaning behind "lazy hydration", in my opinion it implies that the component is still rendered in SSR.

The current implementation is more like a "lazy component until in viewport" 🤔 .

> Though the question here would be, would lazy loading not inherently provide better performance as opposed to lazy hydration?

Probably slightly better with lazy loading. But rendering a static vnode isn't very costly in term of performance.

> There is no reason to make components inside the viewport async regardless, that in and of itself would hurt performance, even if it doesn't use any special wrappers. How can it hurt UX, especially if used properly?

Indeed but some users may use it for a component that is in small breakpoint -> meaning that in desktop it would load the component after the page load resulting in some layout shift.